sonar-db/src/main/java/org/sonar/db/version/v51/FeedFileSourcesBinaryData.java

```diff
@@ -147,7 +148,9 @@ public boolean handle(Select.Row row, SqlStatement update) throws SQLException {
       }
       return FileSourceDto.encodeSourceData(dataBuilder.build());
     } catch (Exception e) {
-      throw new IllegalStateException("Invalid FILE_SOURCES.DATA on row with ID " + fileSourceId + ": " + data, e);
+      LoggerFactory.getLogger(FeedFileSourcesBinaryData.class).error(
+        String.format("Invalid FILE_SOURCES.DATA on row with ID %s, data will be ignored: %s", fileSourceId, data), e);
```

Comment:
Would it be possible to display the functional key instead of the id, so that user knows which projects must be re-analyzed without requesting DB ?